code/modules/events/crystal_event.dm

```diff
@@ -73,8 +76,15 @@ This section is for the event controller
 	dest_crystal = crystal.destabilize(portal_numbers)
 	RegisterSignal(dest_crystal, COMSIG_PARENT_QDELETING, .proc/on_dest_crystal_qdel)
 
-	for(var/turf/turf_loc in RANGE_TURFS(8, dest_crystal.loc))
-		var/distance_from_center = max(get_dist(turf_loc, dest_crystal.loc), 1)
+	for(var/t in RANGE_TURFS(8, dest_crystal.loc))
+		var/turf/turf_loc = t
+		var/distance_from_center = get_dist(turf_loc, dest_crystal.loc)
+		switch(distance_from_center)
+			if(0)
+				distance_from_center = 1 //Same tile, let's avoid a division by zero.
+			if(-1)
+				kill()
+				CRASH("Negative distance error here")
```

Comment:
Give it a bit more informative message. This tells absolutely nothing to someone stumbling upon it.